lib/battle_boats/manual_fleet_deployer.rb

```diff
@@ -0,0 +1,40 @@
+require_relative "console_ui"
+require_relative "board"
+
+module BattleBoats
+  class ManualFleetDeployer
+    def initialize(interface: BattleBoats::ConsoleUI.new,
+                   board: BattleBoats::Board.new(state: :ally))
+      @interface = interface
+      @board = board
+    end
+
+    def deploy(fleet)
+      fleet.ships.each do |ship|
+        place_ship(ship)
+      end
+      @interface.display_board(@board)
+      @board
+    end
+
+    private
+
+    def place_ship(ship)
```

Comment:
How many responsibilities do you think that this method has? I'm curious about the choice of abstraction here as well, although it adheres to the same api as the random class it knows an awful lot about the user interface. Can you think of a way to separate this  from getting input from the user?

Reply:
I think I've counted around 10, and they're mostly all centered around telling `@interface` what to do. 

As far as the abstraction, I think I was so focused on creating a duck that quacks, that I sacrificed taking the time to understand what that quacking should be, so to speak.  (I ended up with a duck that quacks, but this one is also barking, dancing, and selling coffee, too).

In order to decouple, I think I've got to understand _what_ I want this method to accomplish, not _how_.